src/filter.ts

```diff
@@ -110,8 +110,7 @@ export function expression(model: Model, filter: Filter): string {
   if (isString(filter)) {
     return filter;
   } else if (isSelectionFilter(filter)) {
-    const selection = model.getSelectionComponent(filter.selection);
-    return predicate(model, filter.selection, selection.type, selection.resolve, null);
+    return predicate(model, filter.selection);
```

Comment:
Thanks. [Added tests](https://github.com/vega/vega-lite/blob/7c3d1e42e4976be023035182a407bb2fa042690a/test/compile/selection/predicate.test.ts#L83) to ensure that kind of oversight can't happen in the future!